docs/user/reference/hook.md

```diff
@@ -283,14 +285,26 @@ information or advice before signalling the error.
 
 This section describes the workflows and associated hook events which are important to the operation of Juju.
 
-(action-hooks)=
-### Action hooks
+(actions)=
+### Actions
 
-Action hooks operate in an environment with additional environment variables available:
+Actions operate in an execution environment similar to a hook, with additional environment variables available:
 
 * JUJU_ACTION_NAME holds the name of the action.
 * JUJU_ACTION_UUID holds the UUID of the action.
 
+An action is used to perform a named, parameterised operation and report back the results of said operation.
+Actions are defined by the charm and invoked by the user using {ref}`command-juju-run`. The default behaviour is that
+the command blocks, logs any progress messages as reported by the action, and finally prints the action result.
+An action result is a map of key values, containing data set by the action as it runs, plus the overall exit code
+of the action process itself, and the content of stdout and stderr.
+
+The code used to implement an action can the following additional hook commands:
+* action-log: to report a progress message
+* action-get: to get the value of a named action parameter as supplied by the user
+* action-set: to set a value in the action results map 
+* action-fail: to mark the action as failed along with a failure message
+
```

Comment:
We currently list these under hook commands: https://canonical-juju.readthedocs-hosted.com/en/3.6/user/reference/hook-commands/list-of-hook-commands/

Reply:
Technically they are hook commands - they use the same underlying infrastructure and for the unit agent, running an action is essentially the same as running a hook. They also appear when running `juju help-tool`.

Side note: `help-tool` is wrong. It's listed as `Show help on a Juju charm hook tool` but should be reworded as `hook command`.